src/ripple/overlay/impl/Message.cpp

```diff
@@ -109,22 +109,41 @@ Message::compress()
 }
 
 /** Set payload header
- * Uncompressed message header
- * 47-42    Set to 0
- * 41-16    Payload size
- * 15-0	    Message Type
- * Compressed message header
- * 79       Set to 0, indicates the message is compressed
- * 78-76    Compression algorithm, value 1-7. Set to 1 to indicate LZ4
- * compression 75-74    Set to 0 73-48    Payload size 47-32	Message Type
- * 31-0     Uncompressed message size
- */
+
+    The header is a variable-sized structure that contains information about
+    the type of the message and the length and encoding of the payload.
+
+    The first 6 bits determine whether a message is compressed or uncompressed
+    and determine the type of compression:
+
+    All multi-byte values are represented in big endian.
+
+    For uncompressed messages (6 bytes), numbering bits from left to right:
+
+        - The first 6 bits are set to 0.
+        - The next 26 bits represent the payload size.
```

Comment:
Strictly speaking the payload size is 28 bits because we only set 4 bits to 0. Perhaps we should set 6 bits to 0? And use the same rule for the compressed message payload  size and uncompressed message size.